src/util.cpp

```diff
@@ -422,16 +422,14 @@ void ReleaseDirectoryLocks()
 /** Interpret string as boolean, for argument parsing */
 static bool InterpretBool(const std::string& strValue)
 {
-    if (strValue.empty())
-        return true;
-    return (atoi(strValue) != 0);
+    // These values mean false, anything else (including empty string) is true.
```

Comment:
Why not throw an error if it's a stupid value? IMHO, for instance, `-txindex=foo` should fail.

Reply:
Thinking more about this, with proper argument validation we might as well reject `false` as a value for a boolean argument. This was never supported, so it feels weird to special case it now.

Reply:
@promag Two reasons. First, such a change would not be backwards compatible (a minor point, see below). A better reason is that it enables the `-printtoconsole` change to have a way of disabling the debug log file without actually adding new command line arguments. In other words, I first test if `-debuglogfile` boolean parses as "false" (i.e. `-nodebuglogfile` or `-debuglogfile=0`), and if it's not false I re-parse it as a string. There are already a *lot* of possible flags to give Bitcoin, and if we can re-use existing functionality without adding new flags that seems better to me.

@MarcoFalke I actually agree, as accepting `false` opens the door for `disabled`, `untrue`, `nil`, `void`, `zip`, `null`, `non`, `nein`, ad infinitum.

My preferred semantics would be: `-nofoo` sets `foo` to false. Also `-foo=0` sets `foo` to false. Which is exactly what the *intention* of the existing code seems to be. I only handled the `false` flag as there is existing code committed in the Bitcoin repository that relies on it.

I started this endeavor just wanting to get bitcoind printing to stdout when run in interactive mode. Whatever semantics allow that to happen are fine with me.

Reply:
> I first test if -debuglogfile boolean parses as "false" (i.e. -nodebuglogfile or -debuglogfile=0), and if it's not false I re-parse it as a string.

Feel free to discount this opinion, but if the option is `-debuglogfile=<filename>`, I would hope that `-debuglogfile=0` or `-debuglogfile=false` would just create log debug log files `0` or `false` respectively, instead of disabling logging.

Like João and Marco, I would also want ambiguous values passed as boolean options to trigger errors.

I guess I'm not so much a fan of option parsing trying to be very clever.

Reply:
+1 to what @ryanofsky says; if backwards-compatability is not a constraint, doing the least clever thing seems most consistent and likely to lead to the most maintainable code, and long term least confusion for users.

(This PR seems like an improvement in readability of the argument parsing code, so any further simplification might be for another changeset.)